examples/seismic/abc_methods/02_damping.ipynb

```diff
@@ -0,0 +1,1257 @@
+    "        x, z = dimensions\n",
+    "        return {x: ('middle', npmlx, npmlx), z: ('middle', 0, npmlz)}\n",
+    "d0_domain = d0domain()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The blue region will be the union of the following regions:\n",
+    "\n",
+    "- *d1* represents the left range in the direction *x*, where the pairs $(x,z)$ satisfy: $x\\in\\{0,npmlx\\}$ and $z\\in\\{0,nptz\\}$;\n",
+    "- *d2* represents the rigth range in the direction *x*, where the pairs $(x,z)$ satisfy: $x\\in\\{nptx-npmlx,nptx\\}$ and $z\\in\\{0,nptz\\}$;\n",
+    "- *d3* represents the left range in the direction *y*, where the pairs $(x,z)$ satisfy: $x\\in\\{npmlx,nptx-npmlx\\}$ and $z\\in\\{nptz-npmlz,nptz\\}$;\n",
+    "\n",
+    "Thus, the regions *d1*, *d2* and *d3* are described as follows in the language of *subdomains*:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class d1domain(SubDomain):\n",
+    "    name = 'd1'\n",
+    "    def define(self, dimensions):\n",
+    "        x, z = dimensions\n",
+    "        return {x: ('left',npmlx), z: z}\n",
+    "d1_domain = d1domain()\n",
+    "\n",
+    "class d2domain(SubDomain):\n",
+    "    name = 'd2'\n",
+    "    def define(self, dimensions):\n",
+    "        x, z = dimensions\n",
+    "        return {x: ('right',npmlx), z: z}\n",
+    "d2_domain = d2domain()\n",
+    "\n",
+    "class d3domain(SubDomain):\n",
+    "    name = 'd3'\n",
+    "    def define(self, dimensions):\n",
+    "        x, z = dimensions\n",
+    "        return {x: ('middle', npmlx, npmlx), z: ('right',npmlz)}\n",
+    "d3_domain = d3domain()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The figure below represents the division of domains that we did previously:\n",
+    "\n",
+    "<img src='domain2.png' width=500>\n",
+    "\n",
+    "The advantage of dividing into regions is that the equations will be calculated where they actually operate and thus we gain computational efficiency, as we decrease the number of operations to be done. After defining the spatial parameters and constructing the *subdomains*, we set the *spatial grid* with the following command:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "grid = Grid(origin=origin, extent=extent, shape=shape, subdomains=(d0_domain,d1_domain,d2_domain,d3_domain), dtype=np.float64)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Again, we use a velocity field given by a binary file. The reading and scaling of the velocity field for the Devito work units is done with the following commands:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "v0 = np.zeros((nptx,nptz))                     \n",
```

Comment:
This block could use a few comments.

Reply:
What kind of comment would be needed here? We have created an artificial speed field where the user can adapt according to his needs or even use an external example. We ever considered the hole domain, where we included the ABC region.

Reply:
Just brief comments stating what `x10`, `pxm` etc. are.

Reply:
Solved!